waku/v2/discv5/discover.go

```diff
@@ -291,6 +300,9 @@ func evaluateNode(node *enode.Node) bool {
 // used for caching enr address in peerExchange
 // used for connecting to peers in discovery_connector
 func (d *DiscoveryV5) Iterator() (enode.Iterator, error) {
+	d.Lock()
+	defer d.Unlock()
+
```

Comment:
Did not get the purpose of this lock. 
Let's assume that this is to lock when someone is iterating, stop is not called. 

But since we are returning the iterator itself, this lock is not enough to protect caller from using the iterator after stop is called which may lead to invalid access.